sdks/python/apache_beam/io/filesystemio.py

```diff
@@ -157,6 +162,16 @@ def seekable(self):
   def readable(self):
     return True
 
+  def readall(self):
+    """Read until EOF, using multiple read() call."""
+    res = []
+    while True:
```

Comment:
Where is this function used ?

Prob. remove if unused.

Reply:
Ah actually seems like you are overriding the function here: https://docs.python.org/3/library/io.html#io.IOBase

Reply:
Sorry, still have a question. 

Does Beam call readlll() function anywhere ? I couldn't find a usage. Beam textio for example, invokes read() not readall().
https://github.com/apache/beam/blob/master/sdks/python/apache_beam/io/textio.py#L272

If it does, I'm not sure what will prevent us from reading a huge amount of data into memory and running into OOMs.

Reply:
I only found this usage in `ReadableFile` (relatively new) where we don't specify the size:
https://github.com/apache/beam/blob/13825058c1f4368fc7d6fabbab90d3d7d6e9b4e7/sdks/python/apache_beam/io/fileio.py#L150-L154

Reply:
That makes sense. I think ReadableFile is intended for small files. But probably we should add a readall() method there as well and update read() to take a buffer (not in this PR).

cc: @pabloem